File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converters/ConverterDate.java
package pl.szczodrzynski.edziennik.data.db.converters;

import androidx.room.TypeConverter;

import pl.szczodrzynski.edziennik.utils.models.Date;

public class ConverterDate {

    @TypeConverter
    public static Date toDate(String value) {
        return value == null ? null : Date.fromY_m_d(value);
    }

    @TypeConverter
    public static String toString(Date value) {
        return value == null ? null : value.getStringY_m_d();
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterDate.kt
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import pl.szczodrzynski.edziennik.utils.models.Date

class ConverterDate {
    @TypeConverter
    fun toDate(value: String?): Date? = value?.let { Date.fromY_m_d(it) }

    @TypeConverter
    fun toString(value: Date?): String? = value?.stringY_m_d
}
